### PR TITLE
update 3.1.1

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
flask-sqlalchemy 3.1.1

**Destination channel:** defaults

### Links

- [PKG-3393](https://anaconda.atlassian.net/browse/PKG-3393) 
- [Upstream repository](https://github.com/pallets-eco/flask-sqlalchemy)
- [Upstream changelog/diff](https://github.com/pallets-eco/flask-sqlalchemy/releases/tag/3.1.1)

### Explanation of changes:

- flit-core added
- alter url getting

[PKG-3393]: https://anaconda.atlassian.net/browse/PKG-3393?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
   skip: True  # [py<37]
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
   host:
     - python
     - pip
     - pdm-pep517 >=1.0.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
pdm-pep517 is not listed in https://github.com/pallets-eco/flask-sqlalchemy/blob/3.1.1/pyproject.toml#L32

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
-    - setuptools
-    - wheel
+    - flit-core
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - flit-core <4
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
   run:
     - python
     - flask >=2.2
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - flask >=2.2.5
```
https://github.com/pallets-eco/flask-sqlalchemy/blob/3.1.1/pyproject.toml#L19

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,27 +1,26 @@
...
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 16199f5b3ddfb69e0df2f52ae4c76aedbfec823462349dabb21a1b2e0a2b65e9
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ file_name }}-{{ version }}.tar.gz
+  sha256: e4b68bb881802dda1a7d878b2fc84c06d1ee57fb40b874d3dc97dabfa36b8312
 
 build:
   skip: True  # [py<37]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  skip: True  # [py<38]
```
https://github.com/pallets-eco/flask-sqlalchemy/blob/3.1.1/pyproject.toml#L17

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml]: also,` sqlalchemy >=2.0.16`

**Review by cbouss — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,30 +1,28 @@
...
     - pip
-    - pdm-pep517 >=1.0.0
-    - setuptools
-    - wheel
+    - flit-core <4
   run:
     - python
-    - flask >=2.2
+    - flask >=2.2.5
     - sqlalchemy >=1.4.18
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - sqlalchemy >=2.0.16
```
this one too

**Review by cbouss — APPROVED** [review]

**Review by danpetry — APPROVED** [review]

**Pull request merged**